components/frontend/src/header_footer/buttons/DownloadAsPDFButton.test.js

```diff
@@ -0,0 +1,100 @@
+import history from "history/browser"
+
+import * as get_report_pdf from "../../api/report"
+import * as toast from "../../widgets/toast"
+import { DownloadAsPDFButton } from "./DownloadAsPDFButton"
+
+jest.mock("../../api/report")
+jest.mock("../../widgets/toast.js")
+
+beforeEach(() => {
+    jest.resetAllMocks()
+    get_report_pdf.get_report_pdf.mockImplementation(() => Promise.resolve({ ok: true }))
+})
+
+test("DownloadAsPDFButton has the correct label for reports overview", () => {
+    render(<DownloadAsPDFButton />)
+    expect(screen.getAllByLabelText(/reports overview as PDF/).length).toBe(1)
+})
+
+test("DownloadAsPDFButton has the correct label for a report", () => {
+    render(<DownloadAsPDFButton report_uuid={"report_uuid"} />)
+    expect(screen.getAllByLabelText(/report as PDF/).length).toBe(1)
+})
+
+const test_report = { report_uuid: "report_uuid" }
+
+async function clickDownload(nrClicks = 1) {
+    for (let step = 0; step < nrClicks; step++) {
+        await act(async () => {
+            fireEvent.click(screen.getByText(/Download as PDF/))
+        })
+    }
+}
+
+test("DownloadAsPDFButton indicates loading on click", async () => {
+    get_report_pdf.get_report_pdf.mockImplementation(() => {
+        // See https://github.com/eslint-community/eslint-plugin-promise/issues/111#issuecomment-663824152
+        // eslint-disable-line promise/avoid-new
+        return new Promise((resolve) => setTimeout(resolve, 100))
+    })
+    render(<DownloadAsPDFButton report={test_report} report_uuid="report_uuid" />)
+    await clickDownload()
+    expect(screen.getAllByLabelText(/Download/)[0].className).toContain("MuiCircularProgress-indeterminate")
+    expect(get_report_pdf.get_report_pdf).toHaveBeenCalledWith("report_uuid", "?report_url=http%3A%2F%2Flocalhost%2F")
+    await waitFor(() => {
+        expect(toast.showMessage).toHaveBeenCalledTimes(1)
+    })
+})
+
+test("DownloadAsPDFButton ignores unregistered query parameters", async () => {
+    history.push("?unregister_key=value&nr_dates=4")
+    get_report_pdf.get_report_pdf.mockImplementation(() => Promise.resolve({ ok: true }))
+    render(<DownloadAsPDFButton report={test_report} report_uuid="report_uuid" />)
+    await clickDownload()
+    expect(get_report_pdf.get_report_pdf).toHaveBeenCalledWith(
+        "report_uuid",
+        "?nr_dates=4&report_url=http%3A%2F%2Flocalhost%2F%3Fnr_dates%3D4",
+    )
+})
+
+test("DownloadAsPDFButton ignores a second click", async () => {
+    get_report_pdf.get_report_pdf.mockImplementation(() => Promise.resolve({ ok: true }))
+    HTMLAnchorElement.prototype.click = jest.fn() // Prevent "Not implemented: navigation (except hash changes)"
+    window.URL.createObjectURL = jest.fn()
+    render(<DownloadAsPDFButton report={test_report} />)
+    await clickDownload(2)
+    expect(screen.getByLabelText(/Download/).className).not.toContain("MuiCircularProgress-indeterminate")
```

Comment:
I don't get it, does clicking the second time clear the icon?

Reply:
No, this was meant to test that a second click while the PDF is being generated is ignored. But this amounts to testing the MUI button and not our own code, so I'll remove this test.